src/Sulu/Component/Rest/ListBuilder/Doctrine/DoctrineListBuilder.php

```diff
@@ -166,44 +221,95 @@ private function getJoins()
     }
 
     /**
+     * Creates a query-builder for sub-selecting ID's
+     *
+     * @param null|string $select
+     *
+     * @return QueryBuilder
+     */
+    protected function createSubQueryBuilder($select = null)
+    {
+        if (!$select) {
+            $select = $this->entityName . '.id';
+        }
+        $filterFields = array_merge(
+            $this->sortFields,
+            $this->whereFields,
+            $this->inFields,
+            $this->betweenFields,
+            $this->searchFields
+        );
+        // get entity names
+        $filterFields = $this->getEntityNamesOfFieldDescriptors($filterFields);
+
+        // use fields that have filter functionality or have an inner join
+        $addJoins = array();
+        foreach ($this->getJoins() as $entity => $join) {
+            if (array_search($entity, $filterFields) !== false ||
+                $join->getJoinMethod() == DoctrineJoinDescriptor::JOIN_METHOD_INNER
+            ) {
+                $addJoins[$entity] = $join;
+            }
+        }
+
+        $queryBuilder = $this->createQueryBuilder($addJoins)
+            ->select($select);
+
+        return $queryBuilder;
+    }
+
+    /**
+     * Returns filtered list of array of field-descriptors
+     *
+     * @param array $filterFields
+     *
+     * @return string[]
+     */
+    protected function getEntityNamesOfFieldDescriptors($filterFields)
+    {
+        $fields = array();
+
+        // filter array for DoctrineFieldDescriptors
+        foreach ($filterFields as $field) {
+            $fields = array_merge($fields, $field->getJoins());
+            if (!($field instanceof DoctrineConcatenationFieldDescriptor) &&
+                !($field instanceof DoctrineGroupConcatFieldDescriptor)
```

Comment:
Would prefer a whitelist here, since new FieldDescriptors might be added, which could then result in unexpected behavior.